src/spaces/walk.jl

```diff
@@ -125,15 +136,7 @@ function randomwalk!(
         ))
     end
     offsets = offsets_at_radius(model, r)
-    target_positions = map(β -> normalize_position(agent.pos .+ β, model), offsets)
-    idx = findall(pos -> id_in_position(pos, model) == 0, target_positions)
-    available_offsets = @view offsets[idx]
-    if isempty(available_offsets)
-        # don't move, return agent only for type stability
-        agent
-    else
-        walk!(agent, rand(abmrng(model), available_offsets), model; ifempty=false)
-    end
+    walk!(agent, rand(abmrng(model), offsets), model)
```

Comment:
Nicer and faster than before but does not allow us to force the motion of the agent if there available positions. What about adding a switch (`force_motion`?) that allows the user to choose between the two behaviors? That would default to false, giving the new behavior but could be set to true to have the old one.

Reply:
yes, to me this seems the best choice, after all I think that many times one wants to move the agents anyway

Reply:
I'm not sure if we *really* want to have this option I mentioned to choose between the two behaviors, just saying because there might be significant differences in running a simulation one or way or the other.
Also notice that this would be "breaking" for whoever was using this randomwalk function, in the sense that their code would now return something different from before.